// @rn-primitives/table/dist/table.mjs
'use client'

// src/table.tsx
import * as Slot from "@rn-primitives/slot";
import * as React from "react";
import { Pressable as Pressable2, View as View2 } from "react-native";
var Root = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component role='table' ref={ref} {...props} />;
});
Root.displayName = "RootTable";
var Header = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component role='rowheader' ref={ref} {...props} />;
});
Header.displayName = "HeaderTable";
var Row = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Pressable : Pressable2;
  return <Component ref={ref} role='row' {...props} />;
});
Row.displayName = "RowTable";
var Head = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} role='columnheader' {...props} />;
});
Head.displayName = "HeadTable";
var Body = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} role='rowgroup' {...props} />;
});
Body.displayName = "BodyTable";
var Cell = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} role='cell' {...props} />;
});
Cell.displayName = "CellTable";
var Footer = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} role='rowgroup' {...props} />;
});
Footer.displayName = "FooterTable";
export {
  Body,
  Cell,
  Footer,
  Head,
  Header,
  Root,
  Row
};
