// @rn-primitives/popover/dist/popover.web.mjs
'use client'

// src/popover.web.tsx
import * as Popover from "@radix-ui/react-popover";
import { useAugmentedRef, useIsomorphicLayoutEffect } from "@rn-primitives/hooks";
import * as Slot from "@rn-primitives/slot";
import * as React from "react";
import { Pressable as Pressable2, View as View2 } from "react-native";
var RootContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({ asChild, onOpenChange: onOpenChangeProp, ...viewProps }, ref) => {
    const [open, setOpen] = React.useState(false);
    function onOpenChange(value) {
      setOpen(value);
      onOpenChangeProp?.(value);
    }
    const Component = asChild ? Slot.View : View2;
    return <RootContext.Provider value={{ open, onOpenChange }}><Popover.Root open={open} onOpenChange={onOpenChange}><Component ref={ref} {...viewProps} /></Popover.Root></RootContext.Provider>;
  }
);
Root2.displayName = "RootWebPopover";
function useRootContext() {
  const context = React.useContext(RootContext);
  if (!context) {
    throw new Error("Popover compound components cannot be rendered outside the Popover component");
  }
  return context;
}
var Trigger2 = React.forwardRef(
  ({ asChild, onPress: onPressProp, role: _role, disabled, ...props }, ref) => {
    const { onOpenChange, open } = useRootContext();
    const augmentedRef = useAugmentedRef({
      ref,
      methods: {
        open() {
          onOpenChange(true);
        },
        close() {
          onOpenChange(false);
        }
      }
    });
    function onPress(ev) {
      if (onPressProp) {
        onPressProp(ev);
      }
      onOpenChange(!open);
    }
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        augRef.dataset.state = open ? "open" : "closed";
        augRef.type = "button";
      }
    }, [open]);
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Popover.Trigger disabled={disabled ?? void 0} asChild><Component
      ref={augmentedRef}
      onPress={onPress}
      role='button'
      disabled={disabled}
      {...props}
    /></Popover.Trigger>;
  }
);
Trigger2.displayName = "TriggerWebPopover";
function Portal2({ forceMount, container, children }) {
  return <Popover.Portal forceMount={forceMount} children={children} container={container} />;
}
var Overlay = React.forwardRef(
  ({ asChild, forceMount, ...props }, ref) => {
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Component ref={ref} {...props} />;
  }
);
Overlay.displayName = "OverlayWebPopover";
var Content2 = React.forwardRef(
  ({
    asChild = false,
    forceMount,
    align = "start",
    side = "bottom",
    sideOffset = 0,
    alignOffset = 0,
    avoidCollisions = true,
    insets: _insets,
    disablePositioningStyle: _disablePositioningStyle,
    onCloseAutoFocus,
    onEscapeKeyDown,
    onInteractOutside,
    onPointerDownOutside,
    onOpenAutoFocus,
    ...props
  }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <Popover.Content
      onCloseAutoFocus={onCloseAutoFocus}
      onEscapeKeyDown={onEscapeKeyDown}
      onInteractOutside={onInteractOutside}
      onPointerDownOutside={onPointerDownOutside}
      forceMount={forceMount}
      align={align}
      side={side}
      sideOffset={sideOffset}
      alignOffset={alignOffset}
      avoidCollisions={avoidCollisions}
      onOpenAutoFocus={onOpenAutoFocus}
    ><Component ref={ref} {...props} /></Popover.Content>;
  }
);
Content2.displayName = "ContentWebPopover";
var Close2 = React.forwardRef(
  ({ asChild, onPress: onPressProp, disabled, ...props }, ref) => {
    const augmentedRef = useAugmentedRef({ ref });
    const { onOpenChange, open } = useRootContext();
    function onPress(ev) {
      if (onPressProp) {
        onPressProp(ev);
      }
      onOpenChange(!open);
    }
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        augRef.type = "button";
      }
    }, []);
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <><Popover.Close disabled={disabled ?? void 0} asChild><Component
      ref={augmentedRef}
      onPress={onPress}
      role='button'
      disabled={disabled}
      {...props}
    /></Popover.Close></>;
  }
);
Close2.displayName = "CloseWebPopover";
export {
  Close2 as Close,
  Content2 as Content,
  Overlay,
  Portal2 as Portal,
  Root2 as Root,
  Trigger2 as Trigger,
  useRootContext
};
