// @rn-primitives/hover-card/dist/hover-card.web.js
'use client'
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/hover-card.web.tsx
var hover_card_web_exports = {};
__export(hover_card_web_exports, {
  Content: () => Content2,
  Overlay: () => Overlay,
  Portal: () => Portal2,
  Root: () => Root2,
  Trigger: () => Trigger2,
  useRootContext: () => useRootContext
});
module.exports = __toCommonJS(hover_card_web_exports);
var HoverCard = __toESM(require("@radix-ui/react-hover-card"));
var import_hooks = require("@rn-primitives/hooks");
var Slot = __toESM(require("@rn-primitives/slot"));
var React = __toESM(require("react"));
var import_react_native = require("react-native");
var HoverCardContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({ asChild, openDelay, closeDelay, onOpenChange: onOpenChangeProp, ...viewProps }, ref) => {
    const [open, setOpen] = React.useState(false);
    function onOpenChange(value) {
      setOpen(value);
      onOpenChangeProp?.(value);
    }
    const Component = asChild ? Slot.View : import_react_native.View;
    return <HoverCardContext.Provider value={{ open, onOpenChange }}><HoverCard.Root
      open={open}
      onOpenChange={onOpenChange}
      openDelay={openDelay}
      closeDelay={closeDelay}
    ><Component ref={ref} {...viewProps} /></HoverCard.Root></HoverCardContext.Provider>;
  }
);
Root2.displayName = "RootWebHoverCard";
function useRootContext() {
  const context = React.useContext(HoverCardContext);
  if (!context) {
    throw new Error(
      "HoverCard compound components cannot be rendered outside the HoverCard component"
    );
  }
  return context;
}
var Trigger2 = React.forwardRef(({ asChild, ...props }, ref) => {
  const { onOpenChange } = useRootContext();
  const augmentedRef = (0, import_hooks.useAugmentedRef)({
    ref,
    methods: {
      open() {
        onOpenChange(true);
      },
      close() {
        onOpenChange(false);
      }
    }
  });
  const Component = asChild ? Slot.Pressable : import_react_native.Pressable;
  return <HoverCard.Trigger asChild><Component ref={augmentedRef} {...props} /></HoverCard.Trigger>;
});
Trigger2.displayName = "TriggerWebHoverCard";
function Portal2({ forceMount, container, children }) {
  return <HoverCard.Portal forceMount={forceMount} container={container} children={children} />;
}
var Overlay = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Pressable : import_react_native.Pressable;
  return <Component ref={ref} {...props} />;
});
Overlay.displayName = "OverlayWebHoverCard";
var Content2 = React.forwardRef(
  ({
    asChild = false,
    forceMount,
    align,
    side,
    sideOffset,
    alignOffset = 0,
    avoidCollisions = true,
    insets,
    loop: _loop,
    onCloseAutoFocus: _onCloseAutoFocus,
    onEscapeKeyDown,
    onPointerDownOutside,
    onFocusOutside,
    onInteractOutside,
    collisionBoundary,
    sticky,
    hideWhenDetached,
    ...props
  }, ref) => {
    const Component = asChild ? Slot.Pressable : import_react_native.Pressable;
    return <HoverCard.Content
      forceMount={forceMount}
      alignOffset={alignOffset}
      avoidCollisions={avoidCollisions}
      collisionPadding={insets}
      onEscapeKeyDown={onEscapeKeyDown}
      onPointerDownOutside={onPointerDownOutside}
      onFocusOutside={onFocusOutside}
      onInteractOutside={onInteractOutside}
      collisionBoundary={collisionBoundary}
      sticky={sticky}
      hideWhenDetached={hideWhenDetached}
      align={align}
      side={side}
      sideOffset={sideOffset}
    ><Component ref={ref} {...props} /></HoverCard.Content>;
  }
);
Content2.displayName = "ContentWebHoverCard";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Content,
  Overlay,
  Portal,
  Root,
  Trigger,
  useRootContext
});
