// @rn-primitives/menubar/dist/menubar.web.mjs
'use client'

// src/menubar.web.tsx
import * as Menubar from "@radix-ui/react-menubar";
import {
  useAugmentedRef,
  useControllableState,
  useIsomorphicLayoutEffect
} from "@rn-primitives/hooks";
import * as Slot from "@rn-primitives/slot";
import { EmptyGestureResponderEvent } from "@rn-primitives/utils";
import * as React from "react";
import { Pressable as Pressable2, Text as Text2, View as View2 } from "react-native";
var RootContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({ asChild, value, onValueChange, ...viewProps }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <RootContext.Provider value={{ value, onValueChange }}><Menubar.Root value={value} onValueChange={onValueChange}><Component ref={ref} {...viewProps} /></Menubar.Root></RootContext.Provider>;
  }
);
Root2.displayName = "RootWebMenubar";
function useRootContext() {
  const context = React.useContext(RootContext);
  if (!context) {
    throw new Error("Menubar compound components cannot be rendered outside the Menubar component");
  }
  return context;
}
var MenuContext = React.createContext(null);
var Menu2 = React.forwardRef(({ asChild, value, ...viewProps }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <MenuContext.Provider value={{ value }}><Menubar.Menu value={value}><Component ref={ref} {...viewProps} /></Menubar.Menu></MenuContext.Provider>;
});
Menu2.displayName = "MenuWebMenubar";
function useMenuContext() {
  const context = React.useContext(MenuContext);
  if (!context) {
    throw new Error("Menubar compound components cannot be rendered outside the Menubar component");
  }
  return context;
}
var Trigger2 = React.forwardRef(
  ({ asChild, disabled = false, ...props }, ref) => {
    const augmentedRef = useAugmentedRef({ ref });
    const { value: menuValue } = useMenuContext();
    const { value } = useRootContext();
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        augRef.dataset.state = value && menuValue === value ? "open" : "closed";
      }
    }, [value && menuValue]);
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        if (disabled) {
          augRef.dataset.disabled = "true";
        } else {
          augRef.dataset.disabled = void 0;
        }
      }
    }, [disabled]);
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Menubar.Trigger disabled={disabled ?? void 0} asChild><Component ref={augmentedRef} disabled={disabled} {...props} /></Menubar.Trigger>;
  }
);
Trigger2.displayName = "TriggerWebMenubar";
function Portal2({ forceMount, container, children }) {
  return <Menubar.Portal forceMount={forceMount} container={container} children={children} />;
}
var Overlay = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Pressable : Pressable2;
  return <Component ref={ref} {...props} />;
});
Overlay.displayName = "OverlayWebMenubar";
var MenubarContentContext = React.createContext(null);
var Content2 = React.forwardRef(
  ({
    asChild = false,
    forceMount,
    align,
    side,
    sideOffset,
    alignOffset = 0,
    avoidCollisions = true,
    insets,
    loop,
    onCloseAutoFocus,
    onEscapeKeyDown,
    onPointerDownOutside,
    onFocusOutside,
    onInteractOutside,
    collisionBoundary,
    sticky,
    hideWhenDetached,
    ...props
  }, ref) => {
    const itemRef = React.useRef(null);
    function close2() {
      itemRef.current?.click();
    }
    const Component = asChild ? Slot.View : View2;
    return <MenubarContentContext.Provider value={{ close: close2 }}><Menubar.Content
      forceMount={forceMount}
      alignOffset={alignOffset}
      avoidCollisions={avoidCollisions}
      collisionPadding={insets}
      loop={loop}
      onCloseAutoFocus={onCloseAutoFocus}
      onEscapeKeyDown={onEscapeKeyDown}
      onPointerDownOutside={onPointerDownOutside}
      onFocusOutside={onFocusOutside}
      onInteractOutside={onInteractOutside}
      collisionBoundary={collisionBoundary}
      sticky={sticky}
      hideWhenDetached={hideWhenDetached}
      align={align}
      side={side}
      sideOffset={sideOffset}
    ><Component ref={ref} {...props} /><Menubar.Item
      ref={itemRef}
      aria-hidden
      style={{ position: "fixed", top: 0, left: 0, zIndex: -999999999 }}
      aria-disabled
      tabIndex={-1}
      hidden
    /></Menubar.Content></MenubarContentContext.Provider>;
  }
);
Content2.displayName = "ContentWebMenubar";
function useMenubarContentContext() {
  const context = React.useContext(MenubarContentContext);
  if (!context) {
    throw new Error(
      "MenubarContent compound components cannot be rendered outside the MenubarContent component"
    );
  }
  return context;
}
var Item2 = React.forwardRef(
  ({
    asChild,
    textValue,
    closeOnPress = true,
    onPress: onPressProp,
    onKeyDown: onKeyDownProp,
    ...props
  }, ref) => {
    const { close: close2 } = useMenubarContentContext();
    function onKeyDown(ev) {
      onKeyDownProp?.(ev);
      if (ev.key === "Enter" || ev.key === " ") {
        onPressProp?.(EmptyGestureResponderEvent);
        if (closeOnPress) {
          close2();
        }
      }
    }
    function onPress(ev) {
      onPressProp?.(ev);
      if (closeOnPress) {
        close2();
      }
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Menubar.Item
      textValue={textValue}
      disabled={props.disabled ?? void 0}
      onSelect={closeOnPress ? void 0 : onSelected}
      asChild
    ><Component
      ref={ref}
      onKeyDown={onKeyDown}
      onPress={onPress}
      {...props}
    /></Menubar.Item>;
  }
);
Item2.displayName = "ItemWebMenubar";
var Group2 = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Menubar.Group asChild><Component ref={ref} {...props} /></Menubar.Group>;
});
Group2.displayName = "GroupWebMenubar";
var Label2 = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Text : Text2;
  return <Menubar.Label asChild><Component ref={ref} {...props} /></Menubar.Label>;
});
Label2.displayName = "LabelWebMenubar";
var CheckboxItem2 = React.forwardRef(
  ({
    asChild,
    checked,
    onCheckedChange,
    textValue,
    disabled = false,
    closeOnPress = true,
    onPress: onPressProp,
    onKeyDown: onKeyDownProp,
    ...props
  }, ref) => {
    function onKeyDown(ev) {
      onKeyDownProp?.(ev);
      if (ev.key === "Enter" || ev.key === " ") {
        onPressProp?.(EmptyGestureResponderEvent);
        onCheckedChange?.(!checked);
        if (closeOnPress) {
          close();
        }
      }
    }
    function onPress(ev) {
      onPressProp?.(ev);
      onCheckedChange?.(!checked);
      if (closeOnPress) {
        close();
      }
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Menubar.CheckboxItem
      textValue={textValue}
      checked={checked}
      onCheckedChange={onCheckedChange}
      onSelect={closeOnPress ? void 0 : onSelected}
      disabled={disabled ?? void 0}
      asChild
    ><Component
      ref={ref}
      onKeyDown={onKeyDown}
      onPress={onPress}
      role='button'
      {...props}
    /></Menubar.CheckboxItem>;
  }
);
CheckboxItem2.displayName = "CheckboxItemWebMenubar";
var MenubarRadioGroupContext = React.createContext(null);
var RadioGroup2 = React.forwardRef(
  ({ asChild, value, onValueChange, ...props }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <MenubarRadioGroupContext.Provider value={{ value, onValueChange }}><Menubar.RadioGroup value={value} onValueChange={onValueChange} asChild><Component ref={ref} {...props} /></Menubar.RadioGroup></MenubarRadioGroupContext.Provider>;
  }
);
RadioGroup2.displayName = "RadioGroupWebMenubar";
function useMenubarRadioGroupContext() {
  const context = React.useContext(MenubarRadioGroupContext);
  if (!context) {
    throw new Error(
      "MenubarRadioGroup compound components cannot be rendered outside the MenubarRadioGroup component"
    );
  }
  return context;
}
var RadioItem2 = React.forwardRef(
  ({
    asChild,
    value,
    textValue,
    closeOnPress = true,
    onPress: onPressProp,
    onKeyDown: onKeyDownProp,
    ...props
  }, ref) => {
    const { onValueChange } = useMenubarRadioGroupContext();
    const { close: close2 } = useMenubarContentContext();
    function onKeyDown(ev) {
      onKeyDownProp?.(ev);
      if (ev.key === "Enter" || ev.key === " ") {
        onValueChange?.(value);
        onPressProp?.(EmptyGestureResponderEvent);
        if (closeOnPress) {
          close2();
        }
      }
    }
    function onPress(ev) {
      onValueChange?.(value);
      onPressProp?.(ev);
      if (closeOnPress) {
        close2();
      }
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Menubar.RadioItem
      value={value}
      textValue={textValue}
      disabled={props.disabled ?? void 0}
      onSelect={closeOnPress ? void 0 : onSelected}
      asChild
    ><Component
      ref={ref}
      onKeyDown={onKeyDown}
      onPress={onPress}
      {...props}
    /></Menubar.RadioItem>;
  }
);
RadioItem2.displayName = "RadioItemWebMenubar";
var ItemIndicator2 = React.forwardRef(
  ({ asChild, forceMount, ...props }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <Menubar.ItemIndicator forceMount={forceMount} asChild><Component ref={ref} {...props} /></Menubar.ItemIndicator>;
  }
);
ItemIndicator2.displayName = "ItemIndicatorWebMenubar";
var Separator2 = React.forwardRef(
  ({ asChild, decorative, ...props }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <Menubar.Separator asChild><Component ref={ref} {...props} /></Menubar.Separator>;
  }
);
Separator2.displayName = "SeparatorWebMenubar";
var MenubarSubContext = React.createContext(null);
var Sub2 = React.forwardRef(
  ({ asChild, defaultOpen, open: openProp, onOpenChange: onOpenChangeProp, ...props }, ref) => {
    const [open = false, onOpenChange] = useControllableState({
      prop: openProp,
      defaultProp: defaultOpen,
      onChange: onOpenChangeProp
    });
    const Component = asChild ? Slot.View : View2;
    return <MenubarSubContext.Provider value={{ open, onOpenChange }}><Menubar.Sub open={open} onOpenChange={onOpenChange}><Component ref={ref} {...props} /></Menubar.Sub></MenubarSubContext.Provider>;
  }
);
Sub2.displayName = "SubWebMenubar";
function useSubContext() {
  const context = React.useContext(MenubarSubContext);
  if (!context) {
    throw new Error(
      "MenubarSub compound components cannot be rendered outside the MenubarSub component"
    );
  }
  return context;
}
var SubTrigger2 = React.forwardRef(
  ({ asChild, textValue, disabled = false, onPress: onPressProp, ...props }, ref) => {
    const { onOpenChange } = useSubContext();
    function onPress(ev) {
      onOpenChange(true);
      onPressProp?.(ev);
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Menubar.SubTrigger disabled={disabled ?? void 0} textValue={textValue} asChild><Component ref={ref} onPress={onPress} {...props} /></Menubar.SubTrigger>;
  }
);
SubTrigger2.displayName = "SubTriggerWebMenubar";
var SubContent2 = React.forwardRef(
  ({ asChild = false, forceMount, ...props }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <Menubar.Portal><Menubar.SubContent forceMount={forceMount}><Component ref={ref} {...props} /></Menubar.SubContent></Menubar.Portal>;
  }
);
Content2.displayName = "ContentWebMenubar";
function onSelected(ev) {
  ev.preventDefault();
}
export {
  CheckboxItem2 as CheckboxItem,
  Content2 as Content,
  Group2 as Group,
  Item2 as Item,
  ItemIndicator2 as ItemIndicator,
  Label2 as Label,
  Menu2 as Menu,
  Overlay,
  Portal2 as Portal,
  RadioGroup2 as RadioGroup,
  RadioItem2 as RadioItem,
  Root2 as Root,
  Separator2 as Separator,
  Sub2 as Sub,
  SubContent2 as SubContent,
  SubTrigger2 as SubTrigger,
  Trigger2 as Trigger,
  useMenuContext,
  useRootContext,
  useSubContext
};
