// @rn-primitives/dialog/dist/dialog.web.mjs
'use client'

// src/dialog.web.tsx
import * as Dialog from "@radix-ui/react-dialog";
import {
  useAugmentedRef,
  useControllableState,
  useIsomorphicLayoutEffect
} from "@rn-primitives/hooks";
import * as Slot from "@rn-primitives/slot";
import * as React from "react";
import { Pressable as Pressable2, Text as Text2, View as View2 } from "react-native";
var DialogContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({ asChild, open: openProp, defaultOpen, onOpenChange: onOpenChangeProp, ...viewProps }, ref) => {
    const [open = false, onOpenChange] = useControllableState({
      prop: openProp,
      defaultProp: defaultOpen,
      onChange: onOpenChangeProp
    });
    const Component = asChild ? Slot.View : View2;
    return <DialogContext.Provider value={{ open, onOpenChange }}><Dialog.Root open={open} defaultOpen={defaultOpen} onOpenChange={onOpenChange}><Component ref={ref} {...viewProps} /></Dialog.Root></DialogContext.Provider>;
  }
);
Root2.displayName = "RootWebDialog";
function useRootContext() {
  const context = React.useContext(DialogContext);
  if (!context) {
    throw new Error("Dialog compound components cannot be rendered outside the Dialog component");
  }
  return context;
}
var Trigger2 = React.forwardRef(
  ({ asChild, onPress: onPressProp, role: _role, disabled, ...props }, ref) => {
    const augmentedRef = useAugmentedRef({ ref });
    const { onOpenChange, open } = useRootContext();
    function onPress(ev) {
      if (onPressProp) {
        onPressProp(ev);
      }
      onOpenChange(!open);
    }
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        augRef.dataset.state = open ? "open" : "closed";
        augRef.type = "button";
      }
    }, [open]);
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Dialog.Trigger disabled={disabled ?? void 0} asChild><Component
      ref={augmentedRef}
      onPress={onPress}
      role='button'
      disabled={disabled}
      {...props}
    /></Dialog.Trigger>;
  }
);
Trigger2.displayName = "TriggerWebDialog";
function Portal2({ forceMount, container, children }) {
  return <Dialog.Portal forceMount={forceMount} children={children} container={container} />;
}
var Overlay2 = React.forwardRef(
  ({ asChild, forceMount, ...props }, ref) => {
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Dialog.Overlay forceMount={forceMount}><Component ref={ref} {...props} /></Dialog.Overlay>;
  }
);
Overlay2.displayName = "OverlayWebDialog";
var Content2 = React.forwardRef(
  ({
    asChild,
    forceMount,
    onOpenAutoFocus,
    onCloseAutoFocus,
    onEscapeKeyDown,
    onInteractOutside,
    onPointerDownOutside,
    ...props
  }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <Dialog.Content
      onOpenAutoFocus={onOpenAutoFocus}
      onCloseAutoFocus={onCloseAutoFocus}
      onEscapeKeyDown={onEscapeKeyDown}
      onInteractOutside={onInteractOutside}
      onPointerDownOutside={onPointerDownOutside}
      forceMount={forceMount}
    ><Component ref={ref} {...props} /></Dialog.Content>;
  }
);
Content2.displayName = "ContentWebDialog";
var Close2 = React.forwardRef(
  ({ asChild, onPress: onPressProp, disabled, ...props }, ref) => {
    const augmentedRef = useAugmentedRef({ ref });
    const { onOpenChange, open } = useRootContext();
    function onPress(ev) {
      if (onPressProp) {
        onPressProp(ev);
      }
      onOpenChange(!open);
    }
    useIsomorphicLayoutEffect(() => {
      if (augmentedRef.current) {
        const augRef = augmentedRef.current;
        augRef.type = "button";
      }
    }, []);
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <><Dialog.Close disabled={disabled ?? void 0} asChild><Component
      ref={augmentedRef}
      onPress={onPress}
      role='button'
      disabled={disabled}
      {...props}
    /></Dialog.Close></>;
  }
);
Close2.displayName = "CloseWebDialog";
var Title2 = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Text : Text2;
  return <Dialog.Title asChild><Component ref={ref} {...props} /></Dialog.Title>;
});
Title2.displayName = "TitleWebDialog";
var Description2 = React.forwardRef(
  ({ asChild, ...props }, ref) => {
    const Component = asChild ? Slot.Text : Text2;
    return <Dialog.Description asChild><Component ref={ref} {...props} /></Dialog.Description>;
  }
);
Description2.displayName = "DescriptionWebDialog";
export {
  Close2 as Close,
  Content2 as Content,
  Description2 as Description,
  Overlay2 as Overlay,
  Portal2 as Portal,
  Root2 as Root,
  Title2 as Title,
  Trigger2 as Trigger,
  useRootContext
};
