// @rn-primitives/toggle-group/dist/toggle-group.web.js
'use client'
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/toggle-group.web.tsx
var toggle_group_web_exports = {};
__export(toggle_group_web_exports, {
  Item: () => Item2,
  Root: () => Root2,
  useItemContext: () => useItemContext,
  useRootContext: () => useRootContext,
  utils: () => utils
});
module.exports = __toCommonJS(toggle_group_web_exports);
var ToggleGroup = __toESM(require("@radix-ui/react-toggle-group"));
var Slot = __toESM(require("@rn-primitives/slot"));
var import_utils = require("@rn-primitives/utils");
var React = __toESM(require("react"));
var import_react_native = require("react-native");
var ToggleGroupContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({
    asChild,
    type,
    value,
    onValueChange,
    disabled = false,
    rovingFocus,
    orientation,
    dir,
    loop,
    ...viewProps
  }, ref) => {
    const Component = asChild ? Slot.View : import_react_native.View;
    return <ToggleGroupContext.Provider
      value={{
        type,
        value,
        disabled,
        onValueChange
      }}
    ><ToggleGroup.Root
      type={type}
      value={value}
      onValueChange={onValueChange}
      disabled={disabled}
      rovingFocus={rovingFocus}
      orientation={orientation}
      dir={dir}
      loop={loop}
      asChild
    ><Component ref={ref} {...viewProps} /></ToggleGroup.Root></ToggleGroupContext.Provider>;
  }
);
Root2.displayName = "RootToggleGroup";
function useRootContext() {
  const context = React.useContext(ToggleGroupContext);
  if (!context) {
    throw new Error(
      "ToggleGroup compound components cannot be rendered outside the ToggleGroup component"
    );
  }
  return context;
}
var ItemContext = React.createContext(null);
var Item2 = React.forwardRef(
  ({ asChild, value: itemValue, disabled: disabledProp = false, onPress: onPressProp, ...props }, ref) => {
    const { type, disabled, value, onValueChange } = useRootContext();
    function onPress(ev) {
      onPressProp?.(ev);
      if (type === "single") {
        onValueChange(import_utils.ToggleGroupUtils.getNewSingleValue(value, itemValue));
      }
      if (type === "multiple") {
        onValueChange(import_utils.ToggleGroupUtils.getNewMultipleValue(value, itemValue));
      }
    }
    const Component = asChild ? Slot.Pressable : import_react_native.Pressable;
    return <ItemContext.Provider value={{ value: itemValue }}><ToggleGroup.Item value={itemValue} asChild><Component
      ref={ref}
      onPress={onPress}
      disabled={disabled || disabledProp}
      role='button'
      {...props}
    /></ToggleGroup.Item></ItemContext.Provider>;
  }
);
Item2.displayName = "ItemToggleGroup";
function useItemContext() {
  const context = React.useContext(ItemContext);
  if (!context) {
    throw new Error(
      "ToggleGroupItem compound components cannot be rendered outside the ToggleGroupItem component"
    );
  }
  return context;
}
var utils = import_utils.ToggleGroupUtils;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Item,
  Root,
  useItemContext,
  useRootContext,
  utils
});
