// @rn-primitives/navigation-menu/dist/navigation-menu.mjs
'use client'

// src/navigation-menu.tsx
import { useRelativePosition } from "@rn-primitives/hooks";
import { Portal as RNPPortal } from "@rn-primitives/portal";
import * as Slot from "@rn-primitives/slot";
import * as React from "react";
import {
  BackHandler,
  Pressable as Pressable2,
  View as View2
} from "react-native";
var RootContext = React.createContext(null);
var Root = React.forwardRef(
  ({ asChild, value, onValueChange, ...viewProps }, ref) => {
    const nativeID = React.useId();
    const [triggerPosition, setTriggerPosition] = React.useState(null);
    const [contentLayout, setContentLayout] = React.useState(null);
    const Component = asChild ? Slot.View : View2;
    return <RootContext.Provider
      value={{
        value,
        onValueChange,
        nativeID,
        contentLayout,
        setContentLayout,
        setTriggerPosition,
        triggerPosition
      }}
    ><Component ref={ref} role='navigation' {...viewProps} /></RootContext.Provider>;
  }
);
Root.displayName = "RootNativeNavigationMenu";
function useRootContext() {
  const context = React.useContext(RootContext);
  if (!context) {
    throw new Error(
      "NavigationMenu compound components cannot be rendered outside the NavigationMenu component"
    );
  }
  return context;
}
var List = React.forwardRef(({ asChild, ...viewProps }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} role='menubar' {...viewProps} />;
});
List.displayName = "ListNativeNavigationMenu";
var ItemContext = React.createContext(null);
var Item = React.forwardRef(({ asChild, value, ...viewProps }, ref) => {
  const nativeID = React.useId();
  const Component = asChild ? Slot.View : View2;
  return <ItemContext.Provider
    value={{
      value,
      nativeID
    }}
  ><Component ref={ref} role='menuitem' {...viewProps} /></ItemContext.Provider>;
});
Item.displayName = "ItemNativeNavigationMenu";
function useItemContext() {
  const context = React.useContext(ItemContext);
  if (!context) {
    throw new Error(
      "NavigationMenu compound components cannot be rendered outside the NavigationMenu component"
    );
  }
  return context;
}
var Trigger = React.forwardRef(
  ({ asChild, onPress: onPressProp, disabled = false, ...props }, ref) => {
    const triggerRef = React.useRef(null);
    const { value, onValueChange, setTriggerPosition } = useRootContext();
    const { value: menuValue } = useItemContext();
    React.useImperativeHandle(
      ref,
      () => {
        if (!triggerRef.current) {
          return new View2({});
        }
        return triggerRef.current;
      },
      [triggerRef.current]
    );
    function onPress(ev) {
      if (disabled) return;
      triggerRef.current?.measure((_x, _y, width, height, pageX, pageY) => {
        setTriggerPosition({ width, pageX, pageY, height });
      });
      onValueChange(menuValue === value ? "" : menuValue);
      onPressProp?.(ev);
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <Component
      ref={triggerRef}
      aria-disabled={disabled ?? void 0}
      role='button'
      onPress={onPress}
      disabled={disabled ?? void 0}
      aria-expanded={value === menuValue}
      {...props}
    />;
  }
);
Trigger.displayName = "TriggerNativeNavigationMenu";
function Portal({ forceMount, hostName, children }) {
  const navigationMenu = useRootContext();
  const item = useItemContext();
  if (!navigationMenu.triggerPosition) {
    return null;
  }
  if (!forceMount) {
    if (navigationMenu.value !== item.value) {
      return null;
    }
  }
  return <RNPPortal hostName={hostName} name={`${navigationMenu.nativeID}_portal_provider`}><RootContext.Provider
    value={navigationMenu}
    key={`RootContext_${navigationMenu.nativeID}_portal_provider`}
  ><ItemContext.Provider value={item}>{children}</ItemContext.Provider></RootContext.Provider></RNPPortal>;
}
var Content = React.forwardRef(
  ({
    asChild = false,
    forceMount,
    align = "center",
    side = "bottom",
    sideOffset = 0,
    alignOffset = 0,
    avoidCollisions = true,
    onLayout: onLayoutProp,
    insets,
    style,
    disablePositioningStyle,
    ...props
  }, ref) => {
    const {
      value,
      onValueChange,
      triggerPosition,
      setTriggerPosition,
      contentLayout,
      setContentLayout
    } = useRootContext();
    const { value: menuValue, nativeID } = useItemContext();
    React.useEffect(() => {
      const backHandler = BackHandler.addEventListener("hardwareBackPress", () => {
        setTriggerPosition(null);
        setContentLayout(null);
        onValueChange("");
        return true;
      });
      return () => {
        setContentLayout(null);
        backHandler.remove();
      };
    }, []);
    const positionStyle = useRelativePosition({
      align,
      avoidCollisions,
      triggerPosition,
      contentLayout,
      alignOffset,
      insets,
      sideOffset,
      side,
      disablePositioningStyle
    });
    function onLayout(event) {
      setContentLayout(event.nativeEvent.layout);
      onLayoutProp?.(event);
    }
    if (!forceMount) {
      if (value !== menuValue) {
        return null;
      }
    }
    const Component = asChild ? Slot.View : View2;
    return <Component
      ref={ref}
      role='menu'
      nativeID={nativeID}
      aria-modal={true}
      style={[positionStyle, style]}
      onLayout={onLayout}
      onStartShouldSetResponder={onStartShouldSetResponder}
      {...props}
    />;
  }
);
Content.displayName = "ContentNativeNavigationMenu";
var Link = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.Pressable : Pressable2;
  return <Component ref={ref} role='link' {...props} />;
});
Link.displayName = "LinkNativeNavigationMenu";
var Viewport = React.forwardRef((props, ref) => {
  return <View2 ref={ref} {...props} />;
});
Viewport.displayName = "ViewportNativeNavigationMenu";
var Indicator = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <Component ref={ref} {...props} />;
});
Indicator.displayName = "IndicatorNativeNavigationMenu";
function onStartShouldSetResponder() {
  return true;
}
export {
  Content,
  Indicator,
  Item,
  Link,
  List,
  Portal,
  Root,
  Trigger,
  Viewport,
  useItemContext,
  useRootContext
};
