// @rn-primitives/navigation-menu/dist/navigation-menu.web.mjs
'use client'

// src/navigation-menu.web.tsx
import * as NavigationMenu from "@radix-ui/react-navigation-menu";
import { useAugmentedRef, useIsomorphicLayoutEffect } from "@rn-primitives/hooks";
import * as Slot from "@rn-primitives/slot";
import { EmptyGestureResponderEvent } from "@rn-primitives/utils";
import * as React from "react";
import { Pressable as Pressable2, View as View2 } from "react-native";
var NavigationMenuContext = React.createContext(null);
var Root2 = React.forwardRef(
  ({
    asChild,
    value,
    onValueChange,
    delayDuration,
    skipDelayDuration,
    dir,
    orientation,
    ...viewProps
  }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <NavigationMenuContext.Provider value={{ value, onValueChange, orientation }}><NavigationMenu.Root
      value={value}
      onValueChange={onValueChange}
      delayDuration={delayDuration}
      skipDelayDuration={skipDelayDuration}
      dir={dir}
      orientation={orientation}
    ><Component ref={ref} {...viewProps} /></NavigationMenu.Root></NavigationMenuContext.Provider>;
  }
);
Root2.displayName = "RootWebNavigationMenu";
function useRootContext() {
  const context = React.useContext(NavigationMenuContext);
  if (!context) {
    throw new Error(
      "NavigationMenu compound components cannot be rendered outside the NavigationMenu component"
    );
  }
  return context;
}
var List2 = React.forwardRef(({ asChild, ...viewProps }, ref) => {
  const augmentedRef = useAugmentedRef({ ref });
  const { orientation } = useRootContext();
  useIsomorphicLayoutEffect(() => {
    if (augmentedRef.current) {
      const augRef = augmentedRef.current;
      augRef.dataset.orientation = orientation;
    }
  }, [orientation]);
  const Component = asChild ? Slot.View : View2;
  return <NavigationMenu.List asChild><Component ref={ref} {...viewProps} /></NavigationMenu.List>;
});
List2.displayName = "ListWebNavigationMenu";
var ItemContext = React.createContext(null);
var Item2 = React.forwardRef(({ asChild, value, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <ItemContext.Provider value={{ value }}><NavigationMenu.Item value={value} asChild><Component ref={ref} {...props} /></NavigationMenu.Item></ItemContext.Provider>;
});
Item2.displayName = "ItemWebNavigationMenu";
function useItemContext() {
  const context = React.useContext(ItemContext);
  if (!context) {
    throw new Error(
      "NavigationMenu compound components cannot be rendered outside the NavigationMenu component"
    );
  }
  return context;
}
var Trigger2 = React.forwardRef(
  ({ asChild, onPress: onPressProp, disabled = false, onKeyDown: onKeyDownProp, ...props }, ref) => {
    const { value: rootValue, onValueChange } = useRootContext();
    const { value } = useItemContext();
    function onKeyDown(ev) {
      onKeyDownProp?.(ev);
      if (ev.key === " ") {
        onPressProp?.(EmptyGestureResponderEvent);
        onValueChange(value === rootValue ? "" : value);
      }
    }
    function onPress(ev) {
      onPressProp?.(ev);
      onValueChange(value === rootValue ? "" : value);
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <NavigationMenu.Trigger disabled={disabled ?? void 0} asChild><Component
      ref={ref}
      onKeyDown={onKeyDown}
      onPress={onPress}
      {...props}
    /></NavigationMenu.Trigger>;
  }
);
Trigger2.displayName = "TriggerWebNavigationMenu";
function Portal({ children }) {
  return <>{children}</>;
}
var Content2 = React.forwardRef(
  ({
    asChild = false,
    forceMount,
    align: _align,
    side: _side,
    sideOffset: _sideOffset,
    alignOffset: _alignOffset,
    avoidCollisions: _avoidCollisions,
    onLayout: onLayoutProp,
    insets: _insets,
    disablePositioningStyle: _disablePositioningStyle,
    onEscapeKeyDown,
    onPointerDownOutside,
    onFocusOutside,
    onInteractOutside,
    ...props
  }, ref) => {
    const Component = asChild ? Slot.View : View2;
    return <NavigationMenu.Content
      forceMount={forceMount}
      onEscapeKeyDown={onEscapeKeyDown}
      onPointerDownOutside={onPointerDownOutside}
      onFocusOutside={onFocusOutside}
      onInteractOutside={onInteractOutside}
    ><Component ref={ref} {...props} /></NavigationMenu.Content>;
  }
);
Content2.displayName = "ContentWebNavigationMenu";
var Link2 = React.forwardRef(
  ({ asChild, active, onPress: onPressProp, onKeyDown: onKeyDownProp, ...props }, ref) => {
    const { onValueChange } = useRootContext();
    function onKeyDown(ev) {
      onKeyDownProp?.(ev);
      if (ev.key === "Enter" || ev.key === " ") {
        onPressProp?.(EmptyGestureResponderEvent);
        onValueChange("");
      }
    }
    function onPress(ev) {
      onPressProp?.(ev);
      onValueChange("");
    }
    const Component = asChild ? Slot.Pressable : Pressable2;
    return <NavigationMenu.Link active={active} asChild><Component
      ref={ref}
      role='link'
      onKeyDown={onKeyDown}
      onPress={onPress}
      {...props}
    /></NavigationMenu.Link>;
  }
);
Link2.displayName = "LinkWebNavigationMenu";
var Viewport2 = React.forwardRef((props, ref) => {
  return <Slot.View ref={ref} {...props}><NavigationMenu.Viewport /></Slot.View>;
});
Viewport2.displayName = "ViewportWebNavigationMenu";
var Indicator2 = React.forwardRef(({ asChild, ...props }, ref) => {
  const Component = asChild ? Slot.View : View2;
  return <NavigationMenu.Indicator asChild><Component ref={ref} {...props} /></NavigationMenu.Indicator>;
});
Indicator2.displayName = "IndicatorWebNavigationMenu";
export {
  Content2 as Content,
  Indicator2 as Indicator,
  Item2 as Item,
  Link2 as Link,
  List2 as List,
  Portal,
  Root2 as Root,
  Trigger2 as Trigger,
  Viewport2 as Viewport,
  useItemContext,
  useRootContext
};
